src/Integration.Vsix/ErrorList/SonarErrorListDataSource.cs

```diff
@@ -178,16 +155,16 @@ public void RemoveFactory(IIssuesSnapshotFactory factory)
         {
             lock (sinks)
             {
-                factories.Remove(factory);
+                var wasRemoved = factories.Remove(factory);
 
                 foreach (var sink in sinks)
                 {
                     SafeOperation(sink, "RemoveFactory", () => sink.RemoveFactory(factory));
                 }
 
-                if (factory.CurrentSnapshot.Issues.Contains(selectionService.SelectedIssue))
+                if (wasRemoved)
                 {
-                    selectionService.SelectedIssue = null;
+                    NotifyIssuesChanged(factory);
```

Comment:
This is a bit different behavior: RemoveFactory is called when a file is closed, and we will now notify the taggers that the closed file has changed (as they listen to IssuesChanged). In practice this won't make a difference since the taggers filter by current file path, so they won't do anything. 

https://github.com/SonarSource/sonarlint-visualstudio/blob/e5f23e84787d2bf93dac139c700750b23ffeec25/src/IssueViz/Editor/LocationTagging/LocationTagger.cs#L65